src/vmm/src/lib.rs

```diff
@@ -210,17 +215,29 @@ impl VMM {
         let zero_page_addr = GuestAddress(ZEROPG_START);
 
         // Load the kernel into guest memory.
-        let kernel_load = Elf::load(
+        let kernel_load = match Elf::load(
             &self.guest_memory,
             None,
             &mut kernel_image,
             Some(GuestAddress(kernel_cfg.himem_start)),
-        )
-        .map_err(Error::KernelLoad)?;
+        ) {
+            Ok(result) => result,
+            Err(loader::Error::Elf(elf::Error::InvalidElfMagicNumber)) => BzImage::load(
+                &self.guest_memory,
+                None,
+                &mut kernel_image,
+                Some(GuestAddress(kernel_cfg.himem_start)),
+            )
+            .map_err(Error::KernelLoad)?,
+            Err(e) => {
+                return Err(Error::KernelLoad(e));
```

Comment:
It looks fairly easy to test this in the integration test by creating an empty `Tempfile` and then check that Vmm::try_from returns the KernelLoad error. Can you also add a test for this scenario?